### ISSUE TITLE
Ionic App with regular background BLE tasks

### ISSUE THREAD

**Issue by giowild** [transistorsoft/cordova-plugin-background-fetch#151]
Hello Chris, 
this is not an issue, but only an informative question to see if your plugin can fit this kind of BLE use cases.

I originally posted the following question on the ble-central plugin repository:
https://github.com/don/cordova-plugin-ble-central/issues/850

```
Hi all, does someone have already implemented an ionic+cordova only app(no extra Java code) able to perform regularly a background task with a BLE connection?

In particular I would like to perform a BLE connection, connect, fetch and disconnect, once every 15 minutes, in bakground, to a device with a known service UUID, with initial scanForServiceUUID after OS reboot or after Bluetooth adapter state change.
I know there are background cordova plugins (https://github.com/katzer/cordova-plugin-background-mode) but these do not seem to perform background tasks REGULARLY and continously across phone restarts.
```

At first sight, it seems that your "cordova-plugin-background-fetch" fits exacly this use case, but I'm particularly interested in corner cases such as:
- Will the scheduling survive across device reboots?
- Do you think it would be possible to create notification to the user with results of BLE data exchange?
- Do you think it would be possible to update the ionic ngrx store after ble connection fetch?
- How would you deal with the need of regularly fetching and storing **somewhere** the BLE data, and then load it back into the ionic app, when user opens the app?

It would be really great to have a feedback from you. 

Thanks in advance!

**Comment by christocracy** [comment]
On Android it will probably work.  However, for iOS, you'll never get anything close to *"every 15 minutes"*.  You'll be lucky to get an event several times / day, assuming the user actually opens your app daily.

When you *do* get an event, there is no restriction on what you may do with your short amount of background-time.

Here is a screenshot of the demo app:
- Notice there are no events this past weekend (*05-01*, *05-02*) where I did not open the app.
- Notice there is only 1 event for *04-30*.
- If the iOS user does not open your app daily, iOS **stops firing events.**
- If the iOS user opens your app only *infrequently*, iOS will fire events *infrequently*.

![](https://dl.dropbox.com/s/or7no7gl7fg9jr4/Screen%20Shot%202021-05-03%20at%209.46.58%20AM.png?dl=1)

**Comment by dariosalvi78** [comment]
Hi,

sorry for the intrusion. I didn't want to create an issue for this, and it is somewhat related to this issue anyway.
Some time ago, I posted a feature request to the Apache Cordova team for [a feature](https://github.com/apache/cordova/issues/120) very similar to this plugin.
I am wondering if what you have implemented here is basically the same as I posted or not. If it is the same, then I'll cose the issue at the official Cordova repo.

Thanks.
